app/models/deleted_object.rb

```diff
@@ -8,5 +8,10 @@ class DeletedObject < ApplicationRecord
     scope scoped_class.to_s.underscore.pluralize.to_sym, -> { where(object_type: scoped_class) }
   end
 
-  # TODO: cleanup regularly once the owner is destroyed too :) or this will grow uncontrollably
+  scope :service_owner, -> { where.has { owner_type == Service } }
+  scope :service_owner_not_exists, -> { service_owner.joining { owner.of(Service).outer }.where('services.id IS NULL') }
+  scope :service_owner_event_not_exists, lambda {
```

Comment:
Following https://github.com/3scale/porta/pull/941/files#r298992964, this would become `missing_owner_event`.

Reply:
For the events, I like your plan in https://github.com/3scale/porta/pull/941#discussion_r299449778.